packages/expo-font/plugin/src/withFonts.ts

```diff
@@ -0,0 +1,45 @@
+import { ConfigPlugin, createRunOncePlugin } from 'expo/config-plugins';
+import fs from 'fs';
+import path from 'path';
+
+import { withFontsAndroid } from './withFontsAndroid';
+import { withFontsIos } from './withFontsIos';
+
+const pkg = require('expo-font/package.json');
+
+type Font = {
+  alias?: string;
+  fontFamily: string;
+  path: string;
+};
+
+export type FontProps = {
+  fonts?: (Font | string)[];
+};
+
+const withFonts: ConfigPlugin<FontProps> = (config, props) => {
+  const fonts =
+    props.fonts
+      ?.map((p) => {
+        const resolvedPath = path.resolve(
+          config._internal?.projectRoot,
+          typeof p === 'string' ? p : p.path
+        );
+        if (fs.statSync(resolvedPath).isDirectory()) {
+          return fs.readdirSync(resolvedPath).map((file) => path.join(resolvedPath, file));
+        }
```

Comment:
Doing that on Android is no problem because it's a single mod. On iOS there are two that require that resolved paths. We have to do the parsing inside async mod so we'd need to do it three times. Maybe that's ok?